ui/src/components/payload_editor/PayloadEditor.js

```diff
@@ -0,0 +1,71 @@
+import React, {Component} from "react";
+import AceEditor from "react-ace";
+import Row from "react-bootstrap/Row";
+import Col from "react-bootstrap/Col";
+
+import "ace-builds/src-noconflict/mode-json";
+import "ace-builds/src-noconflict/theme-xcode";
+
+export class PayloadEditor extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      didChange: false,
+      newCode: undefined,
```

Comment:
I suggest more precise namings here.
For instance, I can't tell what kind of data `newCode` holds.

Reply:
Removed, since not necessary any longer.